process/block/preprocess/scheduledTxsExecution.go

```diff
@@ -619,15 +628,13 @@ func (ste *scheduledTxsExecution) IsScheduledTx(txHash []byte) bool {
 	return ok
 }
 
-// SetScheduledMiniBlocksAsExecuted sets all the scheduled mini blocks as executed
-func (ste *scheduledTxsExecution) SetScheduledMiniBlocksAsExecuted() {
-	ste.mutScheduledTxs.Lock()
-	defer ste.mutScheduledTxs.Unlock()
+// IsMiniBlockExecuted returns true if the given mini block is already executed
+func (ste *scheduledTxsExecution) IsMiniBlockExecuted(mbHash []byte) bool {
```

Comment:
Yep, this will be done when we will have the mini block header IsFinal property. I will add also here a TODO to be refactored after we have the last version in the elrond-go-core repository

Reply:
Actually we can remove this only when we could differentiate between the final mini blocks executed as scheduled in the last block and the normal mini blocks from the current block which are also final, but not yet executed